Sources/MarathonCore/Create.swift

```diff
@@ -55,20 +55,55 @@ internal final class CreateTask: Task, Executable {
 
             return try editTask.execute()
         }
+      
+        let scriptFile = try createScriptIfNeeded(atPath: path)
+        
+        if arguments.contains("--tests") {
+            try createTests(atPath: path)
+        }
 
+        if !argumentsContainNoOpenFlag {
+            let script = try scriptManager.script(at: scriptFile.path, allowRemote: false)
+            try script.edit(arguments: arguments, open: true)
+        }
+    }
+    
+    private func createScriptIfNeeded(atPath path: String) throws -> File {
+        if arguments.contains("--tests"), let file = try? Folder.current.file(atPath: path) {
+            return file
+        }
+      
         guard let data = makeScriptContent().data(using: .utf8) else {
             throw Error.failedToCreateFile(path)
         }
-
+        
         let file = try perform(FileSystem().createFile(at: path, contents: data),
                                orThrow: Error.failedToCreateFile(path))
-
+        
         printer.output("🐣  Created script at \(path)")
 
-        if !argumentsContainNoOpenFlag {
-            let script = try scriptManager.script(at: file.path, allowRemote: false)
-            try script.edit(arguments: arguments, open: true)
+        return file
+    }
+    
+    @discardableResult private func createTests(atPath path: String) throws -> File {
+        let script = try scriptManager.script(at: path, allowRemote: false)
```

Comment:
Since you already have the script when you are calling this method, how about taking it as an argument instead?